pakyow-core/lib/pakyow/core/routing/route.rb

```diff
@@ -0,0 +1,100 @@
+require "pakyow/support/aargv"
+require "pakyow/core/routing/hook_merger"
+
+module Pakyow
+  module Routing
+    # A route, consisting of a name, path, and pipeline. The pipeline is a list
+    # of procs and/or method names that are called in order when the route is
+    # matched and executed. Pipelines are constructed at runtime.
+    #
+    # @api private
+    class Route
+      include HookMerger
+
+      attr_reader :matcher, :path, :method, :name, :block, :hooks, :pipeline
+
+      def initialize(path_or_matcher, name: nil, method: nil, hooks: {}, &block)
+        @name, @method, @hooks, @block = name, method, hooks, block
+
+        if path_or_matcher.is_a?(String)
+          @path    = path_or_matcher
+          @matcher = create_matcher_from_path(@path)
+        else
+          @path    = ""
+          @matcher = path_or_matcher
+        end
+
+        # TODO: pass input
+        @pipeline = compile_pipeline
+      end
+
+      # TODO: this logic can be shared with router
+      def match(path_to_match)
+        return false unless matcher.match?(path_to_match)
+
+        if matcher.respond_to?(:match)
+          matcher.match(path_to_match).named_captures
+        else
+          true
+        end
+      end
+
+      def call(context)
+        @pipeline.each do |route|
+          if route.is_a?(Proc)
+            context.instance_exec(&route)
+          else
+            context.__send__(route)
```

Comment:
I noticed that sometimes `send` is used and here `__send__`. I prefer `__send__`, but maybe it would be best to only do it one way. Which is your preference?